Tools/autotest/quadplane.py

```diff
@@ -169,6 +169,9 @@ def AirMode(self):
             self.wait_ready_to_arm()
 
         self.start_subtest("Verify that arming with switch does not spin motors in other modes")
+        # disable compass magnetic field arming check that is triggered by the simulated lean of vehicle
```

Comment:
Why should the vehicle lean angle make a difference?  You're translating the magnetic field into body frame, right?

Reply:
@peterbarker,

The issue is that in this auto test the vehicle is not actually being leaned over.  Instead we are tricking the AHRS into thinking it is leaning over by adjusting the AHRS_TRIM_X/Y values.  The vehicle remains flat.. and so the compass values being sent into the arming check remain unchanged.

Reply:
... that can happen on a real vehicle too.  What sort of angle could we have in the trims before the user won't be able to arm?

Reply:
Could we augment this comment, please to say we don't apply trims to the compass?

Reply:
Ok, I've added an additional comment, "# this is required because adjusting the AHRS_TRIM values only affects the IMU and not external compasses"